CMakeLists.txt

```diff
@@ -105,6 +105,8 @@ option(FLB_RELEASE             "Build with release mode (-O2 -g -DNDEBUG)"   No)
 set(FLB_IPO "ReleaseOnly" CACHE STRING "Build with interprocedural optimization")
 set_property(CACHE FLB_IPO PROPERTY STRINGS "On;Off;ReleaseOnly")
 option(FLB_SMALL               "Optimise for small size"       No)
+set(FLB_SECURITY "ReleaseOnly" CACHE STRING "Build with security optimizations")
```

Comment:
My cmake-fu is not strong so can you clarify the expected behaviour here?

I think we need to opt-in to this to keep any existing behaviour and performance. I agree it's good to have the option but we should not enable by default at least initially.

Reply:
By default, this setting will enable these flags during release builds. The performance impact will be very small, since I tried to balance security and performance. For me, it would be fine to start with the setting disabled and enable it in a few months, after users had a chance to try it.